code/io/joy.h

```diff
@@ -35,6 +35,8 @@ namespace io
 			HAT_RIGHTDOWN,
 			HAT_LEFTUP,
 			HAT_LEFTDOWN,
+
+			HAT_NUM_POS     // This one is always last
```

Comment:
If you are using the enum length trick then the first element must have the value 0 or the indices may be invalid.